### PR TITLE
Initial take on Histogram._fillnd

### PR DESCRIPTION
**Pull request by pavoljuhas**
@chrisvam, @ordirules, @ericdill - please review and let me know if you feel it is good to merge.

### CODE REVIEW

File: skxray/core/accumulators/histogram.pyx [hunk 1]
```diff
@@ -121,7 +129,7 @@ class Histogram:
             self._fill2d(coords[0], coords[1], weights)
         else:
             # do the generalized ND histogram
-            raise NotImplementedError()
+            self._fillnd(coords, weights)
```
Comment (ericdill) [inline skxray/core/accumulators/histogram.pyx:130]: This makes me really happy :)  Hooray for nD histogramming!

**Comment by ericdill** [comment]
Tests for 1d and 2d are passing locally. Worth testing the higher dimensionality histogrmaming?

File: skxray/core/accumulators/histogram.pyx [hunk 2]
```diff
@@ -164,28 +172,72 @@ class Histogram:
...
-        '''
+    def _fillnd(self, coords, np.ndarray[wnumtype, ndim=1] weight):
+        cdef np.int_t* aint_ptr[MAX_DIMENSIONS]
+        cdef int aint_stride[MAX_DIMENSIONS]
+        cdef int aint_count = 0
+        cdef np.float_t* afloat_ptr[MAX_DIMENSIONS + 1]
+        cdef int afloat_stride[MAX_DIMENSIONS]
+        cdef int afloat_count = 0
+        istrides = np.asarray(self._values.strides, dtype=np.int32)
+        istrides //= self._values.itemsize
```
Comment (jrmlhermitte) [inline skxray/core/accumulators/histogram.pyx]: Very neat way to get the strides!

File: skxray/core/accumulators/histogram.pyx [hunk 3]
```diff
@@ -164,28 +172,72 @@ class Histogram:
...
+        cdef int [:] dataindexstrides = istrides
+        # determine order of coordinate arrays according to their
+        # numerical data type
+        numtypes = [np.dtype(int), np.dtype(float)]
+        numtypeindex = {tp : i for i, tp in enumerate(numtypes)}
+        ctypeindices = [numtypeindex.get(c.dtype, MAX_DIMENSIONS)
```
Comment (jrmlhermitte) [inline skxray/core/accumulators/histogram.pyx]: should this be len(numtypeindex)+1 and not MAX_DIMENSIONS?

**Comment by jrmlhermitte** [comment]
This is very elegant, good job Pavol! You have taught me a lot of neat tricks in python (which I assumed you also just taught yourself in the past 2 days or so ;-) )

I took a quick look now. I think benchmarking would be a good idea @ericdill  no? I will also try running it on my 2D simulation (basically, go from a 1D detector to 2D detector and do the same sort of thing). Moving via striding is always tricky and often leads to small issues.

**Pull request merged**